plugins/src/lib.rs

```diff
@@ -172,27 +172,29 @@ impl Parse for DeriveSerde {
                 );
                 continue;
             }
-            match meta.lit {
-                Lit::Bool(LitBool { value: true, .. }) if cfg!(feature = "serde1") => {
-                    result = result.and(Ok(Some(true)))
-                }
-                Lit::Bool(LitBool { value: true, .. }) => {
-                    extend_errors!(
+            if let Expr::Lit(expr_lit) = &meta.value {
```

Comment:
what if meta.value does not match Expr::Lit? I think there should be an error in that case.

On HEAD, the following service won't compile, but it does with this change:

```rs
#[tarpc::service(derive_serde = loop {})]
pub trait World {
    /// Returns a greeting for name.
    async fn hello(name: String) -> String;
}
```

Error on HEAD:

```
error: expected literal
  --> example-service/src/lib.rs:12:33
   |
12 | #[tarpc::service(derive_serde = loop {})]
```

Also, I think this means we're missing a test case. Would you like to add one to tarpc/tests/compile_fail? I'm also happy to add it too.

Reply:
Disregard that last question — I added a test case for this in https://github.com/google/tarpc/commit/cef01a1d590923a4577292d49c7a880ac5e17eb1 (feel free to TRYBUILD=overwrite if you update the compile error to be more specific)